src/main/java/net/sf/jabref/logic/util/Version.java

```diff
@@ -0,0 +1,163 @@
+            this.isDevelopmentVersion = false;
+            return;
+        }
+
+        this.fullVersion = version;
+        isDevelopmentVersion = version.contains("dev");
+        String[] versionParts = version.split("dev");
+        String[] versionNumbers = versionParts[0].split(Pattern.quote("."));
+        this.major = Integer.parseInt(versionNumbers[0]);
+        this.minor = versionNumbers.length >= 2 ? Integer.parseInt(versionNumbers[1]) : 0;
+        this.patch = versionNumbers.length >= 3 ? Integer.parseInt(versionNumbers[2]) : 0;
+    }
+
+    /**
+     * Grabs the latest release version from the JabRef GitHub repository
+     *
+     * @return
+     * @throws IOException
+     */
+    public static Version getLatestVersion() throws IOException {
+        URLConnection connection = new URL(JABREF_GITHUB_URL).openConnection();
+        connection.setRequestProperty("Accept-Charset", "UTF-8");
+        BufferedReader rd = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+        JSONObject obj = new JSONObject(rd.readLine());
+        return new Version(obj.getString("tag_name").replaceFirst("v", ""));
+    }
+
+    public static Version getIgnoredVersion(JabRefPreferences preferences) {
+        return new Version(preferences.get(JabRefPreferences.VERSION_IGNORED_UPDATE));
+    }
+
+    /**
+     * @return true if this version is newer than the passed one
+     */
+    public boolean isNewerThan(Version otherVersion) {
+        Objects.requireNonNull(otherVersion);
+        if (Objects.equals(this, otherVersion)) {
+            return false;
+        }
+        if (this.getFullVersion().equals(BuildInfo.UNKNOWN_VERSION)) {
+            return false;
+        }
+        if (otherVersion.getFullVersion().equals(BuildInfo.UNKNOWN_VERSION)) {
+            return true;
+        }
+
+        if (this.getMajor() > otherVersion.getMajor()) {
+            return true;
+        }
+        if (this.getMajor() == otherVersion.getMajor()) {
+            if (this.getMinor() > otherVersion.getMinor()) {
+                return true;
+            }
+            if (this.getMinor() == otherVersion.getMinor() && this.getPatch() > otherVersion.getPatch()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public String getFullVersion() {
+        return fullVersion;
+    }
+
+    public int getMajor() {
+        return major;
+    }
+
+    public int getMinor() {
+        return minor;
+    }
+
+    public int getPatch() {
+        return patch;
+    }
+
+    public boolean isDevelopmentVersion() {
+        return isDevelopmentVersion;
+    }
+
+    /**
+     * @return The link to the changelog on github to this specific version
+     * (https://github.com/JabRef/jabref/blob/vX.X/CHANGELOG.md)
+     */
+    public String getChangelogUrl() {
+        return "https://github.com/JabRef/jabref/blob/v" + this.getMajor() + "." + this.getMinor() + "/CHANGELOG.md";
```

Comment:
The patch number is in the version number only when it's non-zero, right?
Or do you now implement it all the time? e.g. 4.4.0